src/devtools/views/Profiler/utils.js

```diff
@@ -68,3 +75,128 @@ export const scale = (
+        profilingOperationsForOneRoot.push(Array.from(operations));
+      });
+      profilingOperationsForEachRoot.push([
+        rootIDFromQueue,
+        profilingOperationsForOneRoot,
+      ]);
+    }
+
+    const profilingSnapshotForOneRoot = profilingSnapshotsByRootID.get(
+      rootIDFromQueue
+    );
+    if (profilingSnapshotForOneRoot != null) {
+      profilingSnapshotsForEachRoot.push([
+        rootIDFromQueue,
+        profilingSnapshotForOneRoot,
+      ]);
+    }
+  }
+
+  const profilingSummaryForExport: ExportedProfilingSummaryFromFrontend = {
+    version: 1,
+    profilingOperationsForEachRoot,
+    profilingSnapshotsForEachRoot,
+    rendererID,
+    rootID,
+  };
+
+  return profilingSummaryForExport;
+}
+
+function parseExportedProfilingDataV1(
+  data: any /* either ExportedProfilingDataV1 or something else (exception) */
+): ImportedProfilingData {
+  // TODO (profiling) Ensure integrity of all deep structures of ExportedProfilingDataV1 prior to type conversion.
+  if (
+    !(
+      data.version === 1 &&
+      Array.isArray(data.profilingOperations) &&
+      Array.isArray(data.profilingSnapshot) &&
+      Array.isArray(data.commitDetails) &&
+      (typeof data.interactions === 'object' && data.interactions !== null) &&
+      Array.isArray(data.interactions.interactions) &&
+      typeof data.interactions.rootID === 'number' &&
+      (typeof data.profilingSummary === 'object' &&
+        data.profilingSummary !== null)
+    )
+  ) {
+    throw new Error('Malformed profile data version 1.');
+  }
+  const importedProfileData: ImportedProfilingData = {
+    version: data.version,
+    profilingOperations: new Map(
+      data.profilingOperations.map(pair => {
+        if (
+          !(
+            Array.isArray(pair) &&
+            pair.length === 2 &&
+            typeof pair[0] === 'number' &&
+            Array.isArray(pair[1])
+          )
+        ) {
+          throw new Error(
+            'Malformed profile data version 1: profilingOperations pair'
+          );
+        }
+        return [
+          pair[0],
+          pair[1].map(array => {
+            if (
+              !(Array.isArray(array) && array.every(x => typeof x === 'number'))
+            ) {
+              throw new Error(
+                'Malformed profile data version 1: profilingOperations pair[1] array'
+              );
+            }
+            return Uint32Array.from(array);
+          }),
+        ];
+      })
+    ),
+    profilingSnapshot: new Map(data.profilingSnapshot),
+    commitDetails: data.commitDetails,
+    interactions: data.interactions,
+    profilingSummary: data.profilingSummary,
+  };
+  return importedProfileData;
+}
+
+export function parseExportedProfilingDataString(
+  str: string
+): ImportedProfilingData {
+  const jsonData = JSON.parse(str);
+  // When adding newer versions, add a parser into this map.
+  const parsers = new Map([[1, parseExportedProfilingDataV1]]);
```

Comment:
This is an interesting idea. I'm not sure if it's worth doing though. Seems like too many changes wouldn't be backwards compat without complicating the UI.

Reply:
It's okay to keep just one here if you want to. But make the intent very clear using names, not assumed.

Reply:
I never advocated for "assuming" 😄 We need a runtime version check (as the TODO that was in place before called for)

>  // TODO (profiling) Version check; warn if older version.

Reply:
I removed the TODO because I built the version check and some of the deep structure checks.